nodeup/pkg/model/convenience.go

```diff
@@ -27,3 +34,31 @@ func s(v string) *string {
 func i64(v int64) *int64 {
 	return fi.Int64(v)
 }
+
+func getProxyEnvVars(proxies *kops.EgressProxySpec) []v1.EnvVar {
+	glog.Info("in getProxyEnvVars")
+	if proxies == nil {
+		glog.Info("proxies is == nil, returning empty list")
+		return []v1.EnvVar{}
+	}
+
+	if proxies.HTTPProxy.Host == "" {
+		glog.Warning("EgressProxy set but no proxy host provided")
```

Comment:
We can also check this in validation, but no harm in calling it out here.

If we know this is going to go wrong, should we skip setting the env vars?

We could also return an error in this case...